src/Cli/dotnet/Parser.cs

```diff
@@ -153,27 +156,49 @@ internal class CommandLineConsole : IConsole
 
         internal class DotnetHelpBuilder : HelpBuilder
         {
-            public DotnetHelpBuilder(IConsole console, int maxWidth = int.MaxValue) : base(console, Resources.Instance, maxWidth) { }
+            private DotnetHelpBuilder(int maxWidth = int.MaxValue) : base(LocalizationResources.Instance, maxWidth) { }
 
             public static Lazy<HelpBuilder> Instance = new Lazy<HelpBuilder>(() => {
                 int windowWidth;
                 try
                 {
-                    windowWidth = System.Console.WindowWidth;
+                    windowWidth = Console.WindowWidth;
                 }
                 catch
                 {
                     windowWidth = int.MaxValue;
                 }
 
-                DotnetHelpBuilder dotnetHelpBuilder = new DotnetHelpBuilder(new SystemConsole(), windowWidth);
+                DotnetHelpBuilder dotnetHelpBuilder = new DotnetHelpBuilder(windowWidth);
                 dotnetHelpBuilder.Customize(FormatCommandCommon.DiagnosticsOption, defaultValue: Tools.Format.LocalizableStrings.whichever_ids_are_listed_in_the_editorconfig_file);
                 dotnetHelpBuilder.Customize(FormatCommandCommon.IncludeOption, defaultValue: Tools.Format.LocalizableStrings.all_files_in_the_solution_or_project);
                 dotnetHelpBuilder.Customize(FormatCommandCommon.ExcludeOption, defaultValue: Tools.Format.LocalizableStrings.none);
+
+                SetHelpCustomizations(dotnetHelpBuilder);
+
                 return dotnetHelpBuilder;
             });
 
-            public override void Write(ICommand command)
+            private static void SetHelpCustomizations(HelpBuilder builder)
+            {
+                foreach (var option in HelpDescriptionCustomizations.Keys)
+                {
+                    Func<ParseResult, string> descriptionCallback = (ParseResult parseResult) =>
+                    {
+                        foreach (var (command, helpText) in HelpDescriptionCustomizations[option])
```

Comment:
Any perf impact from the nested foreaches?

Reply:
Shouldn't be, this will only be called if we're displaying one of the few help texts that needs a special description customization.